File: matlab/tests/generate_golden_reference/copy_to_golden.m
%% Copy MATLAB outputs from test_output/ to test_reference/

test_output = '../../test_output';
test_reference = '../../test_reference';

% Find all MATLAB output files
datasets = dir(test_output);
datasets = datasets([datasets.isdir] & ~startsWith({datasets.name}, '.'));

fprintf('Copying MATLAB outputs to test_reference/...\n');

for d = 1:length(datasets)
    dataset = datasets(d).name;

    % Find test folders
    tests = dir(fullfile(test_output, dataset));
    tests = tests([tests.isdir] & ~startsWith({tests.name}, '.'));

    for t = 1:length(tests)
        test_name = tests(t).name;
        src_dir = fullfile(test_output, dataset, test_name);
        dst_dir = fullfile(test_reference, dataset, test_name);

        % Find MATLAB files
        files = [dir(fullfile(src_dir, '*_matlab.csv')); ...
                 dir(fullfile(src_dir, '*_matlab.png'))];

        if ~isempty(files)
            % Create destination
            if ~isfolder(dst_dir)
                mkdir(dst_dir);
            end

            % Copy files
            for f = 1:length(files)
                copyfile(fullfile(src_dir, files(f).name), ...
                        fullfile(dst_dir, files(f).name));
                fprintf('  %s/%s/%s\n', dataset, test_name, files(f).name);
            end
        end
    end
end

fprintf('Done!\n');

File: python/tests/generate_golden_reference/copy_to_golden.py
"""Copy Python outputs from test_output/ to test_reference/"""

import shutil
from pathlib import Path

# Setup paths
project_root = Path(__file__).parent.parent.parent.parent
test_output = project_root / 'test_output'
test_reference = project_root / 'test_reference'

print('Copying Python outputs to test_reference/...')

# Find all Python output files
for dataset_dir in test_output.iterdir():
    if not dataset_dir.is_dir() or dataset_dir.name.startswith('.'):
        continue

    for test_dir in dataset_dir.iterdir():
        if not test_dir.is_dir() or test_dir.name.startswith('.'):
            continue

        # Find Python files
        python_files = list(test_dir.glob('*_python.csv')) + \
                      list(test_dir.glob('*_python.png'))

        if python_files:
            # Create destination
            dst_dir = test_reference / dataset_dir.name / test_dir.name
            dst_dir.mkdir(parents=True, exist_ok=True)

            # Copy files
            for src_file in python_files:
                dst_file = dst_dir / src_file.name
                shutil.copy2(src_file, dst_file)
                print(f'  {dataset_dir.name}/{test_dir.name}/{src_file.name}')

print('Done!')
